Sources/Tools/SoundDecompress/main.cpp

```diff
@@ -0,0 +1,258 @@
+                        hsSetBits(flags, plManifestFile::kSndFlagStreamCompressed);
+                    else
+                        hsSetBits(flags, plManifestFile::kSndFlagCacheStereo);
+                }
+
+                std::pair<ST::string, uint16_t> pair = std::make_pair(filename.AsString(), flags);
+                sfxArray.insert(pair);
+            }
+
+            // Unref the object so it goes away
+            buffer->GetKey()->UnRefObject();
+        }
+    }
+
+    soundKeys.clear();
+
+    plIndirectUnloadIterator iter;
+    rm->IterateAllPages(&iter);
+
+    return true;
+}
+
+
+void DecompressSounds(SoundSet& sounds, bool overwrite, OutputStyle verbosity)
+{
+    uint32_t total = sounds.size();
+    uint32_t curr  = 0;
+
+    if (verbosity == kVerbose)
+    {
+        printf("There are %d sounds\n\n", total);
+    }
+
+    for (auto it = sounds.begin(); it != sounds.end(); ++it)
+    {
+        curr++;
+        plFileName path = plFileName::Join("sfx", it->first);
+
+        if (verbosity == kVerbose)
+        {
+            printf("%s\n", path.AsString().c_str());
+        }
+        else if (verbosity == kProgress)
+        {
+            uint32_t percent = (100 * curr) / total;
+            uint32_t progress = uint32_t((float(curr) / total) * 75);
+
+            fprintf(stdout, "\r%3d%% ", percent);
+            for (uint32_t i = 0; i < progress; i++)
+            {
+                fprintf(stdout, "=");
+            }
+            fflush(stdout);
+        }
+
+        if (hsCheckBits(it->second, plManifestFile::kSndFlagCacheSplit))
+        {
+#if OUTPUT_WAV_FILES
+            plOldAudioFileReader::CacheFile(path, true, !overwrite);
+#else
+            plAudioFileReader::CacheFile(path, true, !overwrite);
+#endif
+        }
+        else if (hsCheckBits(it->second, plManifestFile::kSndFlagCacheStereo))
+        {
+#if OUTPUT_WAV_FILES
+            plOldAudioFileReader::CacheFile(path, false, !overwrite);
+#else
+            plAudioFileReader::CacheFile(path, false, !overwrite);
+#endif
+        }
+    }
+
+
+    if (verbosity == kProgress)
+    {
+        // Hack to ensure we always end with 100%
+        fprintf(stdout, "\r100%%\n");
+    }
+}
+
+
+int main(int argc, const char** argv)
+{
+    bool overwrite = false;
+    OutputStyle verbosity = kProgress;
+
+    for (uint32_t arg = 1; arg < argc; arg++)
+    {
+        if (!strcmp(argv[arg], "-h") || !strcmp(argv[arg], "--help"))
+        {
+            PrintHelp();
+            return 0;
+        }
+        else if (!strcmp(argv[arg], "-s") || !strcmp(argv[arg], "--silent"))
+        {
+            verbosity = kSilent;
+        }
+        else if (!strcmp(argv[arg], "-v") || !strcmp(argv[arg], "--verbose"))
+        {
+            verbosity = kVerbose;
+        }
+        else if (!strcmp(argv[arg], "-f") || !strcmp(argv[arg], "--force"))
+        {
+            overwrite = true;
+        }
+    }
```

Comment:
Instead of all this crazy c-string code, can we use the `plCmdParser` that comes with the engine?

Reply:
Yeah probably... this was made by copy-pasting large sections of the `plPageInfo` tool, and trying to maintain a compatible structure for potential OU backporting. But for our copy it makes sense to use what's available.